packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs

```diff
@@ -147,12 +147,19 @@ private MethodProvider BuildCreateRequestMethod(InputOperation operation)
                     .. AppendQueryParameters(uri, operation, paramMap),
                     request.Uri().Assign(uri.ToUri()).Terminate(),
                     .. AppendHeaderParameters(request, operation, paramMap),
+                    .. GetSetContent(request, signature.Parameters),
                     message.Apply(options).Terminate(),
                     Return(message)
                 ]),
                 this);
         }
 
+        private IReadOnlyList<MethodBodyStatement> GetSetContent(ScopedApi<PipelineRequest> request, IReadOnlyList<ParameterProvider> parameters)
+        {
+            var contentParam = parameters.FirstOrDefault(p => ReferenceEquals(p, ScmKnownParameters.BinaryContent));
```

Comment:
any reason that this cannot simply just be
```suggestion
            var contentParam = parameters.FirstOrDefault(p => p == ScmKnownParameters.BinaryContent);
```

Reply:
I like that ReferenceEquals is more explicit what the intent here is.  I do realize they both effectively do the same thing for now.